File: DIYHomeAutomation/app/src/main/java/com/example/diyhomeautomation/models/History.kt
package com.example.diyhomeautomation.models

import java.util.Date

/**
 * Data class representing historical data related to a device.
 *
 * @property id The unique identifier for the history entry.
 * @property value The recorded value associated with the history entry.
 * @property date The date of the recorded data.
 * @property deviceId The ID of the device associated with the history entry.
 *
 * @constructor Creates an instance of History with the provided properties.
 */
data class History(
    var id: Int,
    var value: Double,
    var date: Date,
    var deviceId: Int
)

File: DIYHomeAutomationAPI/DIYHomeAutomationAPI/Models/History.cs
namespace DIYHomeAutomationAPI.Models
{
    /// <summary>
    /// This class represents a History
    /// </summary>
    public class History
    {
        /// <summary>
        /// History's Id
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Device's Value
        /// </summary>
        public double Value { get; set; }

        /// <summary>
        /// History's Date
        /// </summary>
        public DateTime Date { get; set; }

        /// <summary>
        /// Device's Id
        /// </summary>
        public int DeviceId { get; set; }
    }
}
